libsolidity/parsing/Parser.cpp

```diff
@@ -1106,7 +1106,7 @@ ASTPointer<Block> Parser::parseBlock(ASTPointer<ASTString> const& _docString)
 			BOOST_THROW_EXCEPTION(FatalError()); /* Don't try to recover here. */
 		m_inParserRecovery = true;
 	}
-	if (m_parserErrorRecovery)
+	if (m_inParserRecovery)
```

Comment:
Was it a typo?
(BTW, I guess we can replace `m_inParserRecovery` with a local variable, but that's another story.)

Reply:
I don't think it was a typo. `m_parserErrorRecovery` means that we perform recovery at all (instead of just stopping at the first fatal error), while `m_inParserRecovery` means that we are currently trying to find a place to resume parsing at.